public/app/features/panel/metrics_panel_ctrl.ts

```diff
@@ -187,13 +187,18 @@ class MetricsPanelCtrl extends PanelCtrl {
 
     if (this.panel.timeShift) {
       var timeShiftInterpolated = this.templateSrv.replace(this.panel.timeShift, this.panel.scopedVars);
-      var timeShiftInfo = rangeUtil.describeTextRange(timeShiftInterpolated);
+      var timeShiftInfo = rangeUtil.describeTextRange(timeShiftInterpolated.replace(/^-/, ''));
       if (timeShiftInfo.invalid) {
         this.timeInfo = 'invalid timeshift';
         return;
       }
 
-      var timeShift = '-' + timeShiftInterpolated;
+      var timeShift;
+      if (timeShiftInterpolated[0] === '-') {
+        timeShift = timeShiftInterpolated.replace(/^-/, '+');
+      } else {
+        timeShift = '-' + timeShiftInterpolated;
```

Comment:
this whole block seems a bit backwards no? replacing minus with plus.  Why is this done? Can't + be forward timeshift?

Reply:
We could do this but it would require to add a - when the first character is not a +.

Reply:
Is that what you would prefer?

Reply:
- characters should be the default inferred behavior if the interval has no no + or -.